### PR TITLE
PR for Cart Management

### PR DESCRIPTION
**Pull request by SK02K1**
# Cart management added
## Features
- Allow user to add items in cart only if authenticated.
- Navigate user to login page if not authenticated when clicked on add to cart CTA.
- Show cart status on the navbar.
- If an item is already in the cart update the product card CTA from ``` Add to cart ``` to ``` Go to cart ```.
- If there is nothing in the cart, show ``` cart is empty 🛒 ``` message.
- Increment item quantity.
- decrement item quantity.
- Remove item from the cart.
- Private route component to validate user before accessing private routes.

### [Codesandbox](https://codesandbox.io/s/github/SK02K1/E-Commerce/tree/cart-and-wishlist)

Screenshot 2022-03-19 at 10 14 16 PM

Screenshot 2022-03-19 at 10 14 42 PM

### CODE REVIEW

File: src/contexts/cart-context.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          } = await axios.get('/api/user/cart', {
+            headers: {
+              authorization: encodedToken,
+            },
+          });
+          if (statusText === 'OK') {
+            dispatch({ type: CART_ACTIONS.INITIALIZE_CART, payload: { cart } });
+          }
+        } catch (error) {
+          console.log(error.response);
```
Comment (samyakshah3008) [inline src/contexts/cart-context.js:31]: You can remove this console log

File: src/services/cartServices.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    });
+    setIsRemoving(false);
+    if (status === 200) {
+      dispatch({
+        type: CART_ACTIONS.REMOVE_FROM_CART,
+        payload: { updatedCart: cart },
+      });
+    }
+  } catch (error) {
+    console.log(`Error in removing item from cart: ${error.message}`);
```
Comment (samyakshah3008) [inline src/services/cartServices.js:66]: Can remove console log here as well

**Review by Vikas2426 — APPROVED** [review]

**Review by samyakshah3008 — APPROVED** [review]
Clean UI and good naming conventions 
Proper folder structure 

I have given minor suggestion as single comments, you can refactor later 

Loved your documentation of PR Preview 🚀 
Good for approval ✅

**Comment by SK02K1** [comment]
> Clean UI and good naming conventions Proper folder structure
> 
> I have given minor suggestion as single comments, you can refactor later
> 
> Loved your documentation of PR Preview 🚀 Good for approval ✅

Those console logs are just for debugging if an api gives an error. In future they will be replaced by error toast as an enhancement. I should have added this in PR descripition itself 😅 my bad.

**Pull request merged**